client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -1,13 +1,36 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { DatePicker, DatePickerProps } from '@mui/x-date-pickers';
 import { BasicTextInput } from '../../TextInput/BasicTextInput';
 import { StandardTextFieldProps, TextFieldProps } from '@mui/material';
 import { useAppTheme } from '@common/styles';
+import { DateUtils } from '@common/intl';
+import { useDebounceCallback } from '@common/hooks';
 
 export const BaseDatePickerInput: FC<
   Omit<DatePickerProps<Date>, 'renderInput'>
 > = props => {
   const theme = useAppTheme();
+  const [internalValue, setInternalValue] = useState<
+    Date | string | number | null
+  >();
+
+  useEffect(() => {
+    if (props.value) setInternalValue(props.value);
+  }, [props.value]);
+
+  const isInvalid = (value: Date | string | number | null | undefined) => {
+    const dateValue =
+      typeof value === 'string' ? DateUtils.getDateOrNull(value) : value;
+    return !!value && !DateUtils.isValid(dateValue);
+  };
+
+  const debouncedOnChange = useDebounceCallback(
+    value => {
+      if (DateUtils.isValid(value)) props.onChange(value);
+      else setInternalValue(value);
+    },
+    [props.onChange]
+  );
```

Comment:
There are some issues with useDebounceCallback... (

Could you please use something like to rule this out (taken from `packages/programs/src/JsonForms/common/hooks/useDebouncedTextInput.ts`):
```
const debouncedText = useDebouncedValue(text, 500);
  useEffect(() => {
    handleChange(debouncedText);
  }, [debouncedText]);
```

Note, I only realized now that there is already a `useDebouncedValue` implementation in common...

Reply:
Are you actually seeing any specific problems attributable to the `useDebounceCallback` in this case? Cos if so, and we consider it problematic, we should probably make a separate issue to remove it completely and replace it with some of these other things. But that would be quite a big job -- but I was under the impression that it's working fine for most use cases.

Reply:
yes should consider to replace it...

For the text input the issue was when entering text quickly, so yes probably an issue here as well and likely that we run in the race condition as well here.